sw/device/silicon_creator/lib/base/sec_mmio_unittest.cc

```diff
@@ -0,0 +1,189 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+
+#include <array>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "sw/device/silicon_creator/lib/base/mock_abs_mmio.h"
+
+extern "C" {
+// This is an extern in the sec_mmio module.
+sec_mmio_ctx_t sec_mmio_ctx;
+
+// FIXME: Replace for shutdown Mock function.
+uint32_t fake_shutdown_count;
+}
+
+namespace sec_mmio_unittest {
+namespace {
+using ::testing::Each;
+using ::testing::Eq;
+using ::testing::Test;
+
+class SecMmioTest : public mask_rom_test::MaskRomTest {
+ protected:
+  void SetUp() override {
+    *shutdown_count_ = 0;
+    sec_mmio_init(+[] { ++fake_shutdown_count; });
```

Comment:
WDYT about writing the tests that call the shutdown callback as [death tests](https://github.com/google/googletest/blob/master/docs/advanced.md#death-tests)?